pkg/certificate/providers/config_test.go

```diff
@@ -116,3 +119,73 @@ func TestGetCertificateManager(t *testing.T) {
 		})
 	}
 }
+
+func TestGetHashiVaultOSMToken(t *testing.T) {
+	validVaultTokenSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "osm-system",
+			Name:      "osm-vault-token",
+		},
+		Data: map[string][]byte{
+			"token": []byte("token"),
+		},
+	}
+
+	invalidVaultTokenSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "osm-system",
+			Name:      "osm-vault-token",
+		},
+		Data: map[string][]byte{
+			"noop": []byte("noop"),
+		},
+	}
+
+	testCases := []struct {
+		name         string
+		secretKeyRef *v1alpha2.SecretKeyReferenceSpec
+		kubeClient   kubernetes.Interface
+		expectError  bool
+	}{
+		{
+			name: "No Vault token secret",
+			secretKeyRef: &v1alpha2.SecretKeyReferenceSpec{
+				Name:      "osm-vault-token",
+				Namespace: "osm-system",
+				Key:       "token",
+			},
+			kubeClient:  fake.NewSimpleClientset(),
+			expectError: true,
+		},
+		{
+			name: "Invalid Vault token secret",
+			secretKeyRef: &v1alpha2.SecretKeyReferenceSpec{
+				Name:      "osm-vault-token",
+				Namespace: "osm-system",
+				Key:       "token",
+			},
+			kubeClient:  fake.NewSimpleClientset([]runtime.Object{invalidVaultTokenSecret}...),
+			expectError: true,
+		},
+		{
+			name: "Valid Vault token secret",
+			secretKeyRef: &v1alpha2.SecretKeyReferenceSpec{
+				Name:      "osm-vault-token",
+				Namespace: "osm-system",
+				Key:       "token",
+			},
+			kubeClient:  fake.NewSimpleClientset([]runtime.Object{validVaultTokenSecret}...),
+			expectError: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			assert := tassert.New(t)
+
+			token, err := getHashiVaultOSMToken(tc.secretKeyRef, tc.kubeClient)
+			assert.Equal(tc.expectError, token == "")
+			assert.Equal(tc.expectError, err != nil)
```

Comment:
Writing assertions like this is definitely concise, but it's always difficult for me to really understand what they mean then, especially when they fail. e.g.:
```
--- FAIL: TestGetHashiVaultOSMToken (0.00s)
    --- FAIL: TestGetHashiVaultOSMToken/No_Vault_token_secret (0.00s)
        config_test.go:187:
                Error Trace:    config_test.go:187
                Error:          Not equal:
                                expected: false
                                actual  : true
                Test:           TestGetHashiVaultOSMToken/No_Vault_token_secret
        config_test.go:188:
                Error Trace:    config_test.go:188
                Error:          Not equal:
                                expected: false
                                actual  : true
                Test:           TestGetHashiVaultOSMToken/No_Vault_token_secret
FAIL
FAIL    github.com/openservicemesh/osm/pkg/certificate/providers        0.010s
FAIL
```

whereas if this were slightly more verbose, like
```suggestion
			if tc.expectError {
				assert.Empty(token)
				assert.Error(err)
			} else {
				assert.NotEmpty(token)
				assert.NoError(err)
			}
```

then I find it easier to understand:
```
--- FAIL: TestGetHashiVaultOSMToken (0.00s)
    --- FAIL: TestGetHashiVaultOSMToken/No_Vault_token_secret (0.00s)
        config_test.go:191:
                Error Trace:    config_test.go:191
                Error:          Should NOT be empty, but was
                Test:           TestGetHashiVaultOSMToken/No_Vault_token_secret
        config_test.go:192:
                Error Trace:    config_test.go:192
                Error:          Received unexpected error:
                                error retrieving Hashi Vault token secret osm-system/osm-vault-token: secrets "osm-vault-token" not found
                Test:           TestGetHashiVaultOSMToken/No_Vault_token_secret
FAIL
FAIL    github.com/openservicemesh/osm/pkg/certificate/providers        0.010s
FAIL
```